pkg/operator/ceph/cluster/mon/config.go

```diff
@@ -276,8 +276,9 @@ func createClusterAccessSecret(clientset kubernetes.Interface, namespace string,
 	}
 	secret := &v1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      AppName,
-			Namespace: namespace,
+			Name:       AppName,
+			Namespace:  namespace,
+			Finalizers: []string{"kubernetes.io/mon-secrets-protection"},
```

Comment:
How about a rook-specific finalizer name?
```suggestion
			Finalizers: []string{"rook.io/disaster-protection"},
```